ruby/vampires.rb

```diff
@@ -1,30 +1,63 @@
-puts "Enter First Name"
-firstName = gets.chomp
-puts "Enter Middle Initial"
-middleInitial = gets.chomp
-puts "Enter Last Name"
-lastName = gets.chomp
-puts "Enter Age"
-applicantAge = gets.chomp
-puts "Enter Year Born"
-birthYear = gets.chomp
-puts "Would you like some Garlic Bread fresh from our cafeteria?"
-garlicReply = gets.chomp
-puts "Would you like to enroll in the health insurance plan offered by the company?"
-insuranceReply = gets.chomp
-currentYear = Time.new.year
-birthYear - currentYear = ageCheck
-firstName + lastName = fullName
+puts "Welcome to the Werewolf Resources Department"
+puts "How many applicants are being processed at this time?"
+numApplicants = gets.chomp.to_i
+for i in 0..(numApplicants-1)
+  puts "Enter First Name"
+  firstName = gets.chomp
+  puts "Enter Last Name"
+  lastName = gets.chomp
+  puts "Enter Age"
+  applicantAge = gets.chomp
+  puts "Enter Year Born"
+  birthYear = gets.chomp.to_i
+  puts "Would you like some garlic bread fresh from our cafeteria?  Answer y/n"
+  garlicReply = gets.chomp.downcase
+  puts "Would you like to enroll in the health insurance plan? Answer y/n "
+  insuranceReply = gets.chomp.downcase
 
-if ageCheck == applicantAge && (garlicReply == true || insuranceReply == true)
-  puts "Probably not a vampire"
-elsif ageCheck != applicantAge && (garlicReply == false || insuranceReply == false)
-  puts "Probably a vampire"
-elsif ageCheck != applicantAge && (garlicReply ==false && insuranceReply == false)
-  puts "Almost certainly a vampire"
-elsif fullName == "Drake Cula" || fullName == "Tu Fang"
-  puts "Definitely a vampire"
-else
-  puts "Results inconclusive"
+allergyVamp = nil
+
+  loop do
+    puts "Do you have any allergies? When finished, or if no allergies type done"
+    allergyAnswer = gets.chomp.downcase
+    if allergyAnswer == "sunshine"
+      allergyVamp = true
+      break
+    elsif allergyAnswer == "done"
+      break
+    end
+  end
+
+  currentYear = Time.new.year
+  estimatedAge = currentYear - birthYear
+  finalAge = nil
+  fullName = firstName + " " + lastName
+
+  if estimatedAge.to_i == applicantAge.to_i || (estimatedAge.to_i - 1) ==  applicantAge.to_i
+    finalAge = true
+  else
+    finalAge = false
+  end
+
+  puts "Results for #{[firstName, lastName].compact.join(' ')}:"
+  #Comparison of the Applicant's claimed age versus their estimated age
+  #puts "Estimated Age: #{estimatedAge}"
+  #puts "Claimed Age: #{applicantAge}"
+
+  #Formula checks if the Applicant supplied the correct age, accounting   for if applicant's birthday has not happened.
+  if allergyVamp == true
```

Comment:
'Match the following conditions in the order they're listed.' We first read this when starting off the challenge and we may try to hack it out with a basic if/elsif structure but we quickly learn that the first condition to pass will be the result even if a 'more true' version is below.  We may be tempted to switch the order for a working solution, but if we are to follow the constraints we need to be more creative. What if we made each individual scenario it's own respective if block? What would go inside each if block? Surely we can't just puts out the result otherwise our screen would be full of each vampire check that passed. What we could do is keep setting a variable result within each if block and then at the end of all the blocks display the result to look something like this
```
if condition_1
	result = 'low'
end

if condition_2
	result = 'medium'
end

if condition_3
	result = 'high'
end

puts result
```